src/test/java/com/squareup/kotlinpoet/TypeSpecTest.kt

```diff
@@ -474,38 +474,60 @@ class TypeSpecTest {
     val taco = TypeSpec.classBuilder("Taco")
         .addModifiers(KModifier.ABSTRACT)
         .addFun(FunSpec.builder("throwOne")
-            .addException(IOException::class)
+            .addAnnotation(AnnotationSpec.throwsBuilder()
+                .addMember("exceptionClasses", "%T::class", IOException::class.asClassName())
+                .build())
             .build())
         .addFun(FunSpec.builder("throwTwo")
-            .addException(IOException::class)
-            .addException(ClassName(tacosPackage, "SourCreamException"))
+            .addAnnotation(AnnotationSpec.throwsBuilder()
+                .addMember("exceptionClasses", "*arrayOf(%T::class, %T::class)",
+                    IOException::class.asClassName(), ClassName(tacosPackage, "SourCreamException"))
+                .build())
             .build())
         .addFun(FunSpec.builder("abstractThrow")
             .addModifiers(KModifier.ABSTRACT)
-            .addException(IOException::class)
+            .addAnnotation(AnnotationSpec.throwsBuilder()
+                .addMember("exceptionClasses", "%T::class", IOException::class.asClassName())
+                .build())
             .build())
         .addFun(FunSpec.builder("nativeThrow")
             .addModifiers(KModifier.EXTERNAL)
-            .addException(IOException::class)
+            .addAnnotation(AnnotationSpec.throwsBuilder()
+                .addMember("exceptionClasses", "%T::class", IOException::class.asClassName())
+                .build())
             .build())
         .build()
     assertThat(toString(taco)).isEqualTo("""
         |package com.squareup.tacos
         |
         |import java.io.IOException
+        |import kotlin.jvm.Throws
         |
         |abstract class Taco {
-        |  fun throwOne() throws IOException {
+        |  @Throws(
+        |      exceptionClasses = IOException::class
+        |  )
+        |  fun throwOne() {
         |  }
         |
-        |  fun throwTwo() throws IOException, SourCreamException {
+        |  @Throws(
+        |      exceptionClasses = *arrayOf(IOException::class, SourCreamException::class)
```

Comment:
only if we were using the named parameter, switching it to `value` means we don't need to wrap in `*arrayOf()`